src/nvector/serial/nvector_serial.c

```diff
@@ -317,32 +304,25 @@ N_Vector N_VCloneEmpty_Serial(N_Vector w)
 
 N_Vector N_VClone_Serial(N_Vector w)
 {
+  SUNFunctionBegin(w->sunctx);
   N_Vector v;
   sunrealtype* data;
   sunindextype length;
 
   v = NULL;
   v = N_VCloneEmpty_Serial(w);
-  if (v == NULL) { return (NULL); }
+  SUNCheckLastErrNull();
 
   length = NV_LENGTH_S(w);
 
   /* Create data */
-  if (length > 0)
-  {
-    /* Allocate memory */
-    data = NULL;
-    data = (sunrealtype*)malloc(length * sizeof(sunrealtype));
-    if (data == NULL)
-    {
-      N_VDestroy_Serial(v);
-      return (NULL);
-    }
+  data = NULL;
+  data = (sunrealtype*)malloc(length * sizeof(sunrealtype));
+  SUNAssertNull(data, SUN_ERR_MALLOC_FAIL);
```

Comment:
If the input vector `w` has length 0, then shouldn't this malloc and corresponding attach operations still be protected inside `if (length > 0)`?